src/constants/library-info.ts

```diff
@@ -10,3 +10,8 @@ export const TrackingTag = {
   Key: "gu:cdk:version",
   Value: LibraryInfo.VERSION,
 };
+
+export const TrackingTagWithPropagate = {
```

Comment:
I don't think this is correct, as I'm not sure we want to publicly export this variant as it's specifically for the tests and not for "daily use"? In the past we've done something like [this](https://github.com/guardian/cdk/blob/6159612ce108a7e8f4bc9d907896db9ffade42b3/src/constructs/autoscaling/asg.test.ts#L77).

I think there's an opportunity to add a [custom matcher](https://jestjs.io/docs/expect#custom-matchers-api) here (similar to [toHaveResourceOfTypeAndLogicalId](https://github.com/guardian/cdk/blob/73ef8ec96773c53a924e6795286a54565e9ad3ef/src/utils/test/jest.ts#L21)) as it looks like we want to assert a construct has the identity tags. That is, a "toHaveAnIdentityTaggedResource" matcher that'll take a stack, a resource type string and assert that it's tags contain all the [identity keys](https://github.com/guardian/cdk/blob/73ef8ec96773c53a924e6795286a54565e9ad3ef/src/constructs/core/identity.ts#L4-L11)?

Reply:
After discussing this via Chat, we decided to go with the [simpler option](https://github.com/guardian/cdk/pull/399/commits/1e9f5f0831023121684ed17d186e4ae979c9e8a6) for now, in order to get this merged. We'll look at implementing the custom matcher when everyone is around to work on it together.